<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> resumption problem
   </TITLE>
   <LINK REL="Index" HREF="https://mta.openssl.org/pipermail/openssl-users/2020-March/index.html" >
   <LINK REL="made" HREF="mailto:openssl-users%40openssl.org?Subject=Re%3A%20resumption%20problem&In-Reply-To=%3Cf512f17f-e292-4d3a-6430-d935cc8cf4e7%40wizmail.org%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="012067.html">
   <LINK REL="Next"  HREF="012075.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>resumption problem</H1>
    <B>Jeremy Harris</B> 
    <A HREF="mailto:openssl-users%40openssl.org?Subject=Re%3A%20resumption%20problem&In-Reply-To=%3Cf512f17f-e292-4d3a-6430-d935cc8cf4e7%40wizmail.org%3E"
       TITLE="resumption problem">jgh at wizmail.org
       </A><BR>
    <I>Mon Mar 23 23:46:43 UTC 2020</I>
    <P><UL>
        <LI>Previous message: <A HREF="012067.html">Stream filter for CMS_Sign()
</A></li>
        <LI>Next message: <A HREF="012075.html">resumption problem
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#12074">[ date ]</a>
              <a href="thread.html#12074">[ thread ]</a>
              <a href="subject.html#12074">[ subject ]</a>
              <a href="author.html#12074">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>OpenSSL 1.1.1  on Centos 8
Ticket-based resumption


I'm getting a repeatable error from a client call to SSL_connect()
of &quot;14228044:SSL routines:construct_ca_names:internal error&quot;.

Packet capture shows an Alert being sent by the client before
anything is received from the server.

The error only occurs when the client is trying to resume
a previous session, and (here's the odd part) only when
the client is set up to offer a client certificate.

[I can change the client config to stop it offering this
client-cert, and the resumption works just fine]


I *think* possibly also the precise nature of that client cert
matters; a testcase I set up away from my production
system failed to induce the error.  The client cert
is loaded using SSL_CTX_use_certificate_chain_file();
the file contains a private-key and a 3-element chain
with a Lets Encrypt cert (leaf, signer, CA-root).
The CA is sha1/rsa, the other two are sha256/rsa.


The preceding TLS session is logged as using
&quot;TLS1.2:ECDHE-RSA-AES256-GCM-SHA384:256&quot;



Any ideas?
-- 
Thanks,
  Jeremy
</PRE>












<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="012067.html">Stream filter for CMS_Sign()
</A></li>
	<LI>Next message: <A HREF="012075.html">resumption problem
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#12074">[ date ]</a>
              <a href="thread.html#12074">[ thread ]</a>
              <a href="subject.html#12074">[ subject ]</a>
              <a href="author.html#12074">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="../../../mailman/listinfo/openssl-users.html">More information about the openssl-users
mailing list</a><br>
</body></html>
